src/features/ajax/aggregate/index.js

```diff
@@ -15,251 +14,183 @@ import { SUPPORTABILITY_METRIC_CHANNEL } from '../../metrics/constants'
+  #agentRuntime
+  #beacon
+  #proxyBeacon
+  #ajaxMetricDenyListEnabled
+
   constructor (agentIdentifier, aggregator) {
     super(agentIdentifier, aggregator, FEATURE_NAME)
-    const agentInit = getConfiguration(agentIdentifier)
-
-    registerHandler('xhr', storeXhr, this.featureName, this.ee)
-
-    this.waitForFlags(([])).then(() => {
-      const scheduler = new HarvestScheduler('events', {
-        onFinished: onEventsHarvestFinished,
-        getPayload: prepareHarvest
-      }, this)
-      scheduler.startTimer(harvestTimeSeconds)
-      this.drain()
-    })
 
-    const agentRuntime = getRuntime(agentIdentifier)
-    const denyList = agentRuntime.denyList
-    setDenyList(denyList)
+    this.#beacon = getInfo(agentIdentifier).errorBeacon
 
-    let ajaxEvents = []
-    let spaAjaxEvents = {}
-    let sentAjaxEvents = []
-    const ee = this.ee
+    this.#agentRuntime = getRuntime(agentIdentifier)
+    setDenyList(this.#agentRuntime.denyList)
 
+    const agentInit = getConfiguration(agentIdentifier)
+    this.#proxyBeacon = agentInit.proxy?.beacon
+    this.#ajaxMetricDenyListEnabled = agentInit.feature_flags?.includes('ajax_metrics_deny_list')
     const harvestTimeSeconds = agentInit.ajax.harvestTimeSeconds || 10
-    const MAX_PAYLOAD_SIZE = agentInit.ajax.maxPayloadSize || 1000000
+    this.MAX_PAYLOAD_SIZE = agentInit.ajax.maxPayloadSize || 1000000
 
-    // Exposes these methods to browser test files -- future TO DO: can be removed once these fns are extracted from the constructor into class func
-    this.storeXhr = storeXhr
-    this.prepareHarvest = prepareHarvest
-    this.getStoredEvents = function () { return { ajaxEvents, spaAjaxEvents } }
+    this.ajaxEvents = []
+    this.spaAjaxEvents = {}
+    this.sentAjaxEvents = []
+    const classThis = this
 
     // --- v Used by old spa feature
-    ee.on('interactionDone', (interaction, wasSaved) => {
-      if (!spaAjaxEvents[interaction.id]) return
+    this.ee.on('interactionDone', (interaction, wasSaved) => {
+      if (!this.spaAjaxEvents[interaction.id]) return
 
       if (!wasSaved) { // if the ixn was saved, then its ajax reqs are part of the payload whereas if it was discarded, it should still be harvested in the ajax feature itself
-        spaAjaxEvents[interaction.id].forEach(function (item) {
-          ajaxEvents.push(item)
-        })
+        this.spaAjaxEvents[interaction.id].forEach((item) => classThis.ajaxEvents.push(item))
       }
-      delete spaAjaxEvents[interaction.id]
+      delete this.spaAjaxEvents[interaction.id]
     })
     // --- ^
     // --- v Used by new soft nav
-    registerHandler('returnAjax', event => ajaxEvents.push(event), this.featureName, this.ee)
+    registerHandler('returnAjax', event => this.ajaxEvents.push(event), this.featureName, this.ee)
     // --- ^
+    registerHandler('xhr', function () { // the EE-drain system not only switches "this" but also passes a new EventContext with info (horrible practice, I know)
```

Comment:
```suggestion
    registerHandler('xhr', function () { // the EE-drain system not only switches "this" but also passes a new EventContext with info. Should consider platform refactor to another system which passes a mutable context around separately and predictably to avoid problems like this.
```